File: exercises/RB101_small_problems/easy_2/mutation.rb
array1 = %w(Moe Larry Curly Shemp Harpo Chico Groucho Zeppo)
array2 = []
array1.each { |value| array2 << value }
array1.each { |value| value.upcase! if value.start_with?('C', 'S') }
puts array2
# uh oh, array2 values have also changed
# to do this in a better way, use #map! method, and remove '!' from upcase to avoid changes in array2:

array1.map { |value| array2 << value }
array1.map! { |value| value.upcase if value.start_with?('C', 'S') }

File: exercises/JS210_Fundementals/Objects/mutation.js
const array1 = ['Moe', 'Larry', 'Curly', 'Shemp', 'Harpo', 'Chico', 'Groucho', 'Zeppo'];
const array2 = [];

for (let i = 0; i < array1.length; i += 1) {
  array2.push(array1[i]);
}

for (let i = 0; i < array1.length; i += 1) {
  if (array1[i].startsWith('C')) {
    array1[i] = array1[i].toUpperCase();
  }
}

console.log(array2);

// [
//   'Moe',     'Larry',
//   'Curly',   'Shemp',
//   'Harpo',   'Chico',
//   'Groucho', 'Zeppo'
// ]

// To make the changes appear in array2 we must assign array2 = array1 when declaring it